packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -441,19 +459,61 @@ export class SDKClient {
       Hbar.fromTinybars(Math.floor(networkGasPriceInTinyBars * constants.MAX_GAS_PER_SEC)),
     );
 
-    return {
-      fileId,
-      txResponse: await this.executeTransaction(
+    let txResponse;
+    try {
+      txResponse = await this.executeTransaction(
         ethereumTransaction,
         callerName,
         interactingEntity,
         requestDetails,
         true,
         originalCallerAddress,
-      ),
+      );
+    } catch (e: any) {
+      if (e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded())) {
+        const isFailed = await this.isFailedTransaction(requestDetails, e.transactionId);
```

Comment:
An alternative approach is to simply call the `await this.mirrorNodeClient.repeatedRequest(...)` here instead. 
It makes the logic a positive approach and might avoid the need for an extra method in here.

Reply:
@Nana-EC Extracting the logic in a separate method and naming it properly helps the code to be more readable in my opinion. Now if someone checks this code they know why we call the mirror node, namely to check if this transaction has failed. Thats my motivataion for this, tell me if you don't agree.

Reply:
@Nana-EC  Extracting the logic in a separate method and naming it properly helps the code to be more readable in my opinion. Now if someone checks this code they know why we call the mirror node, namely to check if this transaction has failed. Thats my motivataion for this, tell me if you don't agree.

Reply:
Sounds good